### PR TITLE
Prepare framework for `use_super_parameters`

### PR DESCRIPTION
**Pull request by goderbauer**
~BLOCKED on https://github.com/dart-lang/sdk/issues/48769.~ Fixed.
~BLOCKED ON https://github.com/dart-lang/sdk/issues/48727.~ Fixed.
~BLOCKED ON https://github.com/dart-lang/sdk/issues/48719.~ Fixed.
~BLOCKED ON https://github.com/dart-lang/sdk/issues/48697.~ Fixed.

https://github.com/flutter/flutter/issues/100575

Once all code in the repo has been migrated, we can turn on the use_super_parameters lint.

### CODE REVIEW

File: packages/flutter/lib/src/foundation/diagnostics.dart [hunk 1]
```diff
@@ -1844,31 +1844,20 @@ class StringProperty extends DiagnosticsProperty<String> {
...
-    String? value, {
-    String? description,
-    String? tooltip,
-    bool showName = true,
-    Object? defaultValue = kNoDefaultValue,
+    String super.name,
+    super.value, {
+    super.description,
+    super.tooltip,
+    super.showName,
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/diagnostics.dart:1851]: This line produces the following error:

```dart
error: The type 'dynamic' of this parameter isn't a subtype of the type 'bool' of the associated super constructor parameter. (super_formal_parameter_type_is_not_subtype_of_associated at [flutter] lib/src/foundation/diagnostics.dart:1851)
```

This error happens in a couple of places in the framework after migrating to super params. So far, I have been unable to reproduce this outside of the framework.

@pq or @johnniwinther any idea what may be going on here?

Comment (goderbauer) [reply packages/flutter/lib/src/foundation/diagnostics.dart:1851]: I was able to reproduce this and filed https://github.com/dart-lang/sdk/issues/48697.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

RSLGTM: GitHub can't handle this many files in a review, but I assume that they're all basically the same.

**Comment by goderbauer** [comment]
"Google testing" has some kind of infra problem and it's not able to run the tests. I am going to merge this and watch for post-submit failures.

**Pull request merged**